src/App.tsx

```diff
@@ -1,26 +1,44 @@
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import { Loader, TodoFilter, TodoList, TodoModal } from './components';
+import { useEffect, useState } from 'react';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { setTodos } from './features/todos';
+import { getTodos } from './api';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+export const App = () => {
+  const dispatch = useAppDispatch();
+  const [isLoading, setIsLoading] = useState(false);
+  const currentTodo = useAppSelector(state => state.currentTodo);
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos().then(data => {
+      dispatch(setTodos(data));
+      setIsLoading(false);
+    });
+  }, [dispatch]);
+
+  return (
+    <>
+      <div className="section">
+        <div className="container">
+          <div className="box">
+            <h1 className="title">Todos:</h1>
+
+            <div className="block">
+              <TodoFilter />
+            </div>
 
-          <div className="block">
-            <Loader />
-            <TodoList />
+            <div className="block">
+              {isLoading && <Loader />}
+              <TodoList />
```

Comment:
Ensure that the `TodoList` component does not require any props. If it does, you need to pass them here. Otherwise, this might lead to unexpected behavior.